diff --git a/node_modules/@opcat-labs/cat-sdk/dist/cjs/features/cat20/send/singleSend.js b/node_modules/@opcat-labs/cat-sdk/dist/cjs/features/cat20/send/singleSend.js
index d1b9a13..1e7f9bb 100644
--- a/node_modules/@opcat-labs/cat-sdk/dist/cjs/features/cat20/send/singleSend.js
+++ b/node_modules/@opcat-labs/cat-sdk/dist/cjs/features/cat20/send/singleSend.js
@@ -38,6 +38,19 @@ async function singleSend(signer, provider, minterScriptHash, inputTokenUtxos, r
     const pubkey = await signer.getPublicKey();
     const feeChangeAddress = await signer.getAddress();
     let feeUtxos = await provider.getUtxos(feeChangeAddress);
+
+
+    const guardContract = new cat20Guard_1.CAT20Guard();
+    const cat20Contract = new cat20_1.CAT20(minterScriptHash, contractPeripheral_1.ContractPeripheral.scriptHash(guardContract.lockingScript.toHex()));
+    const cat20Script = cat20Contract.lockingScript.toHex();
+    const cat20ScriptHash = contractPeripheral_1.ContractPeripheral.scriptHash(cat20Script);
+    scrypt_ts_opcat_1.assert(cat20ScriptHash == inputTokenUtxos[0].script || cat20Script == inputTokenUtxos[0].script, 'Input token UTXOs do not match the CAT-20 contract script hash.');
+    if (cat20ScriptHash == inputTokenUtxos[0].script) { 
+        inputTokenUtxos.forEach(utxo => {
+            utxo.script = cat20Script;
+        })
+    }
+    
     const { guardPsbt, outputTokenStates, changeTokenOutputIndex, guard } = await singleSendStep1(provider, feeUtxos, inputTokenUtxos, receivers, feeChangeAddress, tokenChangeAddress, feeRate);
     const signedGuardPsbt = scrypt_ts_opcat_1.ExtPsbt.fromHex(await signer.signPsbt(guardPsbt.toHex(), guardPsbt.psbtOptions()));
     guardPsbt.combine(signedGuardPsbt).finalizeAllInputs();
@@ -63,6 +76,7 @@ async function singleSendStep1(provider, feeUtxos, inputTokenUtxos, receivers, f
     if (feeUtxos.length === 0) {
         throw new Error('Insufficient satoshis input amount');
     }
+    
     receivers = [...receivers];
     const inputTokenStates = inputTokenUtxos.map((utxo) => cat20_1.CAT20.deserializeState(utxo.data));
     const totalInputAmount = inputTokenStates.reduce((acc, state) => acc + state.amount, 0n);
diff --git a/node_modules/@opcat-labs/cat-sdk/dist/esm/features/cat20/send/singleSend.js b/node_modules/@opcat-labs/cat-sdk/dist/esm/features/cat20/send/singleSend.js
index 84b78eb..6fe21fa 100644
--- a/node_modules/@opcat-labs/cat-sdk/dist/esm/features/cat20/send/singleSend.js
+++ b/node_modules/@opcat-labs/cat-sdk/dist/esm/features/cat20/send/singleSend.js
@@ -12,6 +12,21 @@ export async function singleSend(signer, provider, minterScriptHash, inputTokenU
     const pubkey = await signer.getPublicKey();
     const feeChangeAddress = await signer.getAddress();
     let feeUtxos = await provider.getUtxos(feeChangeAddress);
+
+
+    const guardContract = new CAT20Guard();
+    const cat20Contract = new CAT20(minterScriptHash, ContractPeripheral.scriptHash(guardContract.lockingScript.toHex()));
+    const cat20Script = cat20Contract.lockingScript.toHex();
+    const cat20ScriptHash = ContractPeripheral.scriptHash(cat20Script);
+    if (cat20ScriptHash !== inputTokenUtxos[0].script && cat20Script !== inputTokenUtxos[0].script) {
+        throw new Error('Input token UTXOs do not match the CAT-20 contract script hash.');
+    }
+    if (cat20ScriptHash == inputTokenUtxos[0].script) {
+        inputTokenUtxos.forEach(utxo => {
+            utxo.script = cat20Script;
+        })
+    }
+
     const { guardPsbt, outputTokenStates, changeTokenOutputIndex, guard } = await singleSendStep1(provider, feeUtxos, inputTokenUtxos, receivers, feeChangeAddress, tokenChangeAddress, feeRate);
     const signedGuardPsbt = ExtPsbt.fromHex(await signer.signPsbt(guardPsbt.toHex(), guardPsbt.psbtOptions()));
     guardPsbt.combine(signedGuardPsbt).finalizeAllInputs();
@@ -36,6 +51,7 @@ export async function singleSendStep1(provider, feeUtxos, inputTokenUtxos, recei
     if (feeUtxos.length === 0) {
         throw new Error('Insufficient satoshis input amount');
     }
+
     receivers = [...receivers];
     const inputTokenStates = inputTokenUtxos.map((utxo) => CAT20.deserializeState(utxo.data));
     const totalInputAmount = inputTokenStates.reduce((acc, state) => acc + state.amount, 0n);
